src/insight/insightClient.ts

```diff
@@ -476,6 +483,121 @@ export class CoveoInsightClient {
+        );
+    }
+
+    public logDislikeGeneratedAnswer(generatedAnswerMetadata: GeneratedAnswerBaseMeta, metadata?: CaseMetadata) {
+        return this.logCustomEvent(
+            SearchPageEvents.dislikeGeneratedAnswer,
+            metadata
+                ? {...generateMetadataToSend(metadata, false), ...generatedAnswerMetadata}
+                : generatedAnswerMetadata
+        );
+    }
+
+    public logOpenGeneratedAnswerSource(
+        generatedAnswerSourceMetadata: GeneratedAnswerCitationMeta,
+        metadata?: CaseMetadata
+    ) {
+        return this.logCustomEvent(
+            SearchPageEvents.openGeneratedAnswerSource,
+            metadata
+                ? {...generateMetadataToSend(metadata, false), ...generatedAnswerSourceMetadata}
+                : generatedAnswerSourceMetadata
+        );
+    }
+
+    public logGeneratedAnswerSourceHover(
+        generatedAnswerSourceMetadata: GeneratedAnswerSourceHoverMeta,
+        metadata?: CaseMetadata
+    ) {
+        return this.logCustomEvent(
+            SearchPageEvents.generatedAnswerSourceHover,
+            metadata
+                ? {...generateMetadataToSend(metadata, false), ...generatedAnswerSourceMetadata}
+                : generatedAnswerSourceMetadata
+        );
+    }
+
+    public logGeneratedAnswerCopyToClipboard(
+        generatedAnswerMetadata: GeneratedAnswerBaseMeta,
+        metadata?: CaseMetadata
+    ) {
+        return this.logCustomEvent(
+            SearchPageEvents.generatedAnswerCopyToClipboard,
+            metadata
+                ? {...generateMetadataToSend(metadata, false), ...generatedAnswerMetadata}
+                : generatedAnswerMetadata
+        );
+    }
+
+    public logGeneratedAnswerHideAnswers(generatedAnswerMetadata: GeneratedAnswerBaseMeta, metadata?: CaseMetadata) {
+        return this.logCustomEvent(
+            SearchPageEvents.generatedAnswerHideAnswers,
+            metadata
+                ? {...generateMetadataToSend(metadata, false), ...generatedAnswerMetadata}
+                : generatedAnswerMetadata
+        );
+    }
+
+    public logGeneratedAnswerShowAnswers(generatedAnswerMetadata: GeneratedAnswerBaseMeta, metadata?: CaseMetadata) {
+        return this.logCustomEvent(
+            SearchPageEvents.generatedAnswerShowAnswers,
+            metadata
+                ? {...generateMetadataToSend(metadata, false), ...generatedAnswerMetadata}
+                : generatedAnswerMetadata
+        );
+    }
+
+    public logGenerativeQuestionFeedbackSubmit(
+        generatedAnswerFeedbackMetadata: GeneratedAnswerFeedbackMeta,
+        metadata?: CaseMetadata
+    ) {
+        return this.logCustomEvent(
+            SearchPageEvents.generativeQuestionFeedbackSubmit,
+            metadata
+                ? {...generateMetadataToSend(metadata, false), ...generatedAnswerFeedbackMetadata}
+                : generatedAnswerFeedbackMetadata
+        );
+    }
+
+    public logRephraseGeneratedAnswer(
```

Comment:
```suggestion
    public logGeneratedAnswerRephrase(
```
To fit with all the others?

Reply:
I don't think we have a rule about that, looking at all the events we currently have we are using these two patterns interchangeably, thus I don't think this change is worth the trouble especially that  both `logRephraseGeneratedAnswer` and `logGeneratedAnswerRephrase` make sense.